pyrobosim/pyrobosim/core/yaml_utils.py

```diff
@@ -225,11 +231,11 @@ def to_dict(self, world):
         }
 
         # Extract the location and object metadata.
-        loc_metadata_file = world.get_location_metadata().filename or ""
-        obj_metadata_file = world.get_object_metadata().filename or ""
+        loc_metadata_files = world.get_location_metadata().sources["locations"]
+        obj_metadata_files = world.get_object_metadata().sources["objects"]
```

Comment:
It seems redundant to have to call e.g. `get_location_metadata()` and then extract a `"locations"` field -- any way to streamline this?